packages/mds-cache/index.ts

```diff
@@ -214,6 +221,18 @@ async function readKeys(pattern: string) {
   return (await getClient()).keysAsync(pattern)
 }
 
+async function getMostRecentEventByProvider(): Promise<{ provider_id: string; max: number }[]> {
+  const provider_ids = (await readKeys('provider:*:latest_event')).map(key => {
+    const [, provider_id] = key.split(':')
+    return provider_id
+  })
+  const result = await hreads(['latest_event'], provider_ids, 'provider')
+  return result.map(elem => {
+    const max = parseInt(elem.recorded || '0')
```

Comment:
Are we looking for the most recently `recorded` event, or the most recent event by `timestamp`?

Reply:
That's a good question. I'm not actually clear on the differences in usage, @avatarneil any input on this?

Reply:
Pretty sure timestamp is what you want. _when the event happened_ as opposed to _when the event was reported/recorded_